docs/docs/03-syntax-and-usage/11-script-templates.md

```diff
@@ -103,3 +103,50 @@ func main() {
 	}
 }
 ```
+
+You can also directly render the Javascript function, passing in Go data, using the `!` expression:
+
+```templ
+package main
+
+script printToConsole(content string) {
+	console.log(conent)
+}
+
+templ page(content string) {
+	<html>
+		<body>
+		  {! printToConsole(content) }
```

Comment:
The `{! templComponent() }` syntax is old syntax that (not relevant here) doesn't support the use of child elements.

In the case of this design, there's no need for child templ elements, so maybe the new syntax of `@printToConsole(content)` would be good to use?

@joerdav has been going through and removing all the references to the old syntax. https://github.com/a-h/templ/pull/280 - but we still need to support the legacy syntax.

In terms of the implementation, it took me a couple of readthroughs to realise that you'd simply implemented the `templ.Component` interface on the `ScriptComponent`. Very smart!

Reply:
Hmm... so swapping it to `@` in the tests seems to have broken it: 

```go
package testscriptusage

script withParameters(a string, b string, c int) {
	console.log(a, b, c);
}

script withoutParameters() {
	alert("hello");
}

templ InlineJavascript(a string) {
	{ @withoutParameters() }
	{ @withParameters(a, "test", 123) }
	// Call once more, to ensure it's defined only once
	{ @withoutParameters() }
	{ @withParameters(a, "test", 123) }
}
```

```
$ go run ./cmd/templ generate -include-version=false && go test ./...
failed to process path: ./templ/generator/test-script-inline/template.templ source formatting error: 43:36: illegal character U+0040 '@' (and 7 more errors)
```

I'll try and see what needs to be fixed to allow the newer syntax :thinking:

Reply:
Never mind :facepalm: I realized `{ @function() }` should be simply `@function()`